src/OmnichannelChatSDK.ts

```diff
@@ -194,66 +196,220 @@ class OmnichannelChatSDK {
-        this.scenarioMarker.startScenario(TelemetryEvent.InitializeChatSDK);
+        console.error("AMSClient is not ready after retries");
+        return null;
+    }
 
-        if (this.isInitialized) {
-            this.scenarioMarker.completeScenario(TelemetryEvent.InitializeChatSDK);
-            return this.liveChatConfig;
+    private async getAMSClient() {
+
+        if (this.AMSClientLoadCurrentState === AMSClientLoadStates.NOT_LOADED && this.liveChatVersion === LiveChatVersion.V1) {
+            return null;
         }
 
-        this.useCoreServicesOrgUrlIfNotSet();
+        switch (this.AMSClientLoadCurrentState) {
+            case AMSClientLoadStates.LOADED:
+                return this.AMSClient;
 
-        const useCoreServices = isCoreServicesOrgUrl(this.omnichannelConfig.orgUrl);
-        try {
-            this.OCSDKProvider = OCSDKProvider;
-            this.OCClient = await OCSDKProvider.getSDK(this.omnichannelConfig as IOmnichannelConfiguration, createOcSDKConfiguration(useCoreServices) as ISDKConfiguration, this.ocSdkLogger as OCSDKLogger);
-            setOcUserAgent(this.OCClient, this.chatSDKConfig?.ocUserAgent);
-        } catch (e) {
-            exceptionThrowers.throwOmnichannelClientInitializationFailure(e, this.scenarioMarker, TelemetryEvent.InitializeChatSDK);
-        }
+            case AMSClientLoadStates.LOADING:
+                return await this.retryLoadAMSClient();
 
-        try {
-            const { getLiveChatConfigOptionalParams } = optionalParams;
-            await this.getChatConfig(getLiveChatConfigOptionalParams || {});
-        } catch (e) {
-            exceptionThrowers.throwChatConfigRetrievalFailure(e, this.scenarioMarker, TelemetryEvent.InitializeChatSDK);
+            case AMSClientLoadStates.ERROR:
+            case AMSClientLoadStates.NOT_LOADED:
+                await this.loadAmsClient();
+                return this.AMSClient;
+
+            default:
+                return null;
         }
 
+    }
+
+    private async loadInitComponents() {
+        this.scenarioMarker.startScenario(TelemetryEvent.InitializeComponents);
+
         const supportedLiveChatVersions = [LiveChatVersion.V1, LiveChatVersion.V2];
         if (!supportedLiveChatVersions.includes(this.liveChatVersion)) {
-            exceptionThrowers.throwUnsupportedLiveChatVersionFailure(new Error(ChatSDKErrorName.UnsupportedLiveChatVersion), this.scenarioMarker, TelemetryEvent.InitializeChatSDK);
+            exceptionThrowers.throwUnsupportedLiveChatVersionFailure(new Error(ChatSDKErrorName.UnsupportedLiveChatVersion), this.scenarioMarker, TelemetryEvent.InitializeComponents);
         }
 
+        this.isInitialized = true;
+        this.scenarioMarker.completeScenario(TelemetryEvent.InitializeComponents);
+    }
+
+    private async loadAmsClient() {
+        this.scenarioMarker.startScenario(TelemetryEvent.InitializeAMSClient);
```

Comment:
Would `InitializeMessagingClient` event make more sense? The method is initialization `ACSClient` & `AMSClient` which are part of the messaging platform we're using.